cmd/server/server.go

```diff
@@ -133,7 +138,17 @@ func runServer(sa *serverArgs) error {
 	adapterMgr := adapterManager.NewManager(adapter.Inventory(), aspect.Inventory(), eval)
 	configManager := config.NewManager(eval, adapterMgr.AspectValidatorFinder(), adapterMgr.BuilderValidatorFinder(),
 		sa.globalConfigFile, sa.serviceConfigFile, time.Second*time.Duration(sa.configFetchIntervalSec))
-	handler := api.NewHandler(adapterMgr, adapterMgr.AspectMap())
+
+	var handler api.Handler
+	if sa.workerPoolSize == 0 {
+		handler = api.NewHandler(adapterMgr, adapterMgr.AspectMap())
+	} else {
+		poolsize := int(sa.workerPoolSize)
+		if poolsize < 0 {
```

Comment:
We cast the uint to an int, so it's possible that the interpretation as bits is a negative number. i.e. uint64 max = 18446744073709551615, int64 max = 9223372036854775807